src/main/java/producer/SerializedProducerRecord.java

```diff
@@ -0,0 +1,48 @@
+package producer;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import commons.header.Header;
+import commons.headers.Headers;
+import commons.serializers.OptionalSerializer;
+import commons.serializers.HeaderSerializer;
+import commons.serializers.HeadersSerializer;
+import commons.serializers.ProducerRecordSerializer;
+
+import java.util.Optional;
+
+public class SerializedProducerRecord {
+    private static Kryo kryo;
+    static final int INITIAL_BUFFER_SIZE = 4096;
+
+    static {
+        // All classes will share a single Kryo object.
+        kryo = new Kryo();
+
+        // Kyro requires each class to be registered for serialization. (Includes nested objects)
+        kryo.register(Optional.class, new OptionalSerializer());
+        kryo.register(Headers.class, new HeadersSerializer());
+        kryo.register(Header.class, new HeaderSerializer());
+    }
+
+    public static <K,V> byte[] serialize(ProducerRecord<K,V> record, Class<K> keyClass, Class<V> valueClass) {
+        // Dynamically register the generic types
+        kryo.register(ProducerRecord.class, new ProducerRecordSerializer<>(keyClass, valueClass));
+
+        final Output output = new Output(INITIAL_BUFFER_SIZE, -1); // 4 KB is the typical size of memory page
+        kryo.writeObject(output, record);
+        output.close();
+        return output.getBuffer();
```

Comment:
is there a way to just get the occupied space of this buffer? i.e., if the buffer is 4096 bytes but only the first 20 bytes are actual serialized data, can we just get those 20 bytes instead? It would be more convenient in that case instead of sending over the entire page which may contain large empty space